File: app/javascript/channels/message_channel.js
// import consumer from "./consumer"

// const Message = consumer.subscriptions.create("MessageChannel", {
//   connected() {
//     // Called when the subscription is ready for use on the server
//   },

//   disconnected() {
//     // Called when the subscription has been terminated by the server
//   },

//   received(data) {
//     return alert(data['message']);
//   },

//   speak: function(message) {
//     console.log(message)
//     console.log('speak js');
//     return this.perform('speak', {message: message});
//   },
// });

// window.addEventListener('keypress', function(e) {
//   if (e.keyCode === 13 ){
//     console.log(e.target.value);
//     Message.speak(e.target.value);
//     e.target.value = '';
//     e.preventDefault();
//   }
// });

// import consumer from "./consumer"

// consumer.subscriptions.create("MessageChannel", {
//   received(data) {
//     // 以下で使用
//     console.log(data);
//     const html = `<p>${data.content.id} | ${data.content.text}</p>`;
//     const messages = document.getElementById('messages');
//     const newMessage = document.getElementById('message_text');
//     messages.insertAdjacentHTML('afterbegin', html);
//     newMessage.value='';
//   }
// });

File: app/channels/message_channel.rb
class MessageChannel < ApplicationCable::Channel
  def subscribed
    p 'channel'
    p 'start sub'
    stream_from "message_channel"

    ActionCable.server.broadcast 'message_channel', messages: Message.all, event: 'all_messages'
  end

  def unsubscribed
    p 'this is unsub'     
  end

  def speak(data)
    p 'speak rb'
    p data

    @message = Message.create(
      text: data['message'],
      # ID１が管理者、それ以外をユーザーとする
      user_id: data['user_id'],
    )
    
    ActionCable.server.broadcast 'message_channel', message: @message, event: 'single_message'
  end
end